qcfractal/tests/test_collections.py

```diff
@@ -110,17 +141,66 @@ def test_gradient_dataset_get_records(gradient_dataset_fixture):
         ds.get_records(method="NotInDataset")
 
 
-def test_gradient_dataset_get_values_no_match(gradient_dataset_fixture):
+def test_gradient_dataset_get_values(gradient_dataset_fixture):
     client, ds = gradient_dataset_fixture
 
-    with pytest.raises(KeyError):
-        ds.get_values(method="NotInDataset")
+    cols = set(ds.get_values().columns.str.replace(' (contributed)', '', regex=False))
+    names = set(ds.list_values().reset_index()['name'])
+    assert cols == names
+
+    df = ds.get_values()
+    assert df.shape == (len(ds.get_index()), 4)
+    for entry in ds.get_index():
+        assert (df.loc[entry, "HF/sto-3g"] == ds.get_records(subset=entry, method="hf", basis="sto-3g", projection={'return_result': True})).all()
+
+    assert ds.get_values(method="NotInDataset").shape[1] == 0  # 0-length DFs can cause exceptions
+
+    with pytest.warns(RuntimeWarning):
+        ds.get_values(name="HF/sto-3g", basis="sto-3g")
+
+
+def test_gradient_dataset_list_values(gradient_dataset_fixture):
+    client, ds = gradient_dataset_fixture
+
+    # List contributed values
+    df = ds.list_values(native=False).reset_index()
+    assert df.shape == (3, 7)
+    assert set(df.columns) == {*ds.data.history_keys, "name", "native"}
+    assert set(map(lambda x: x.lower(), df['name'])) == set(ds.data.contributed_values.keys())
+
+    df = ds.list_values(driver="graDieNt", native=False).reset_index()
+    assert set(map(lambda x: x.lower(), df['name'])) == set(["gradient", "all details"])
```

Comment:
`{x.lower() for x in df['name']}`, slightly shorter and uses simpler comprehension. Also recommend using literals: `{"gradient", "all details"}`.